stock_picking_propagate_scheduled_date/models/stock_move.py

```diff
@@ -24,10 +24,14 @@ def _propagate_date(self, new_date):
                     continue
                 if move_dest.id in already_propagated_ids:
                     continue
-                move_dest.date -= delta
+                move_dest.with_context(skip_date_propagation=True).write(
+                    {
+                        "date": move_dest.date - delta,
+                    }
+                )
```

Comment:
Since the context is already updated at line 16, I'd put this context key in there. This way, the context will be updated once for the whole recordset, instead of being updated singularly on each record.

However, a `RecursionError` looks strange.
Dates are propagated top-bottom using the `stock.move.move_dest_ids` field, which is a `Many2many` field.
I suppose that chaining the moves through this field may result in a circular structure, eg:
```
>>> self
stock.move(1,)
>>> self.move_dest_ids
stock.move(2,)
>>> self.move_dest_ids.move_dest_ids
stock.move(1,)
```
but when calling `self.date = <some_value>`, the propagation should stop at `stock.move(2,)`, since `stock.move(1,)`'s ID is already included in context's `date_propagation_ids` value (so condition at line 25 should be `True`).

Can you explain why this behavior didn't work, and what caused the `RecursionError`?